### PR TITLE
docs: fix TS plugin documentation

### PR DESCRIPTION
**Pull request by lundibundi**
Refs: https://github.com/fastify/help/issues/251

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
lgtm

**Review by fox1t — APPROVED** [review]
LGTM

File: fastify.d.ts [hunk 1]
```diff
@@ -119,7 +119,7 @@ type TrustProxyFunction = (address: string, hop: number) => boolean
 /* Export all additional types */
 export { FastifyRequest, RequestGenericInterface } from './types/request'
 export { FastifyReply } from './types/reply'
-export { FastifyPluginCallback, FastifyPluginAsync, FastifyPluginOptions, FastifyPlugin } from './types/plugin'
+export { FastifyPluginOptions, FastifyPlugin } from './types/plugin'
```
Comment (L2jLiga) [inline fastify.d.ts:122]: @mcollina, @fox1t this breaks `fastify-plugin`.
Also it's the breaking change cause it affects public APIs for TS users :(

image

Comment (lundibundi) [reply fastify.d.ts:122]: I'm not sure I understand the message (it suggests to import the same thing), but could you verify that you are using most recent `fastify-plugin` version? (it should be fixed in 2.0.1 ->https://github.com/fastify/fastify-plugin/commit/d81e6a6b32effc6607e2e7b58c4365d1ec60dc61)

Comment (L2jLiga) [reply fastify.d.ts:122]: I use 2.3.4

Comment (L2jLiga) [reply fastify.d.ts:122]: It was released 4 hours ago

Comment (lundibundi) [reply fastify.d.ts:122]: Oh, now I see the issue.
I don't know how these got removed. I'll put a PR to get them back

Comment (L2jLiga) [reply fastify.d.ts:122]: Thanks!

Comment (lundibundi) [reply fastify.d.ts:122]: Done https://github.com/fastify/fastify/pull/2555.

**Pull request merged**